datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -17,38 +17,46 @@
 public class BookController {
 
     private final BookRepository repository;
-
     private final BookModelAssembler assembler;
+    private final BookStoreRepository storeRepository;
 
-    public BookController(BookRepository repository, BookModelAssembler assembler) {
+    public BookController(BookRepository repository, BookModelAssembler assembler, BookStoreRepository storeRepository) {
         this.repository = repository;
         this.assembler = assembler;
+        this.storeRepository = storeRepository;
     }
 
-    @PostMapping("/books")
-    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book){
+    @PostMapping("/bookstores/{storeID}/books")
+    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book, @PathVariable Long storeID){
+        BookStore store = storeRepository.getById(storeID);
+        book.setStoreID(storeID);
+        book.setStore(store);
         EntityModel<Book> entityModel = assembler.toModel(repository.save(book));
+
         return ResponseEntity
                 .created(entityModel.getRequiredLink(IanaLinkRelations.SELF).toUri())
                 .body(entityModel);
     }
-    
-    @GetMapping("/books/{id}")
-    protected EntityModel<Book> one(@PathVariable Long id) {
+
+    @GetMapping("/bookstores/{storeID}/books/{id}")
+    protected EntityModel<Book> one(@PathVariable Long id, @PathVariable Long storeID) {
         Book book = repository.findById(id).orElseThrow(() -> new BookNotFoundException(id));
```

Comment:
two questions here:

1) what if the book doesnt match the store?
2) what if the storeID doesnt exist?